File: GeneratedEntityReaderClasses/Components/PhysicsShapeComponent.cs
namespace GeneratedNoitaClasses;
public class PhysicsShapeComponent : NoitaComponentBase,  iNoitaType<PhysicsShapeComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool recreate { get; set; } = new();
	public NoitaBool is_circle { get; set; } = new();
	public NoitaBool is_box { get; set; } = new();
	public NoitaBool is_capsule { get; set; } = new();
	public NoitaBool is_based_on_sprite { get; set; } = new();
	public NoitaFloat friction { get; set; } = new();
	public NoitaFloat restitution { get; set; } = new();
	public NoitaFloat density { get; set; } = new();
	public NoitaFloat local_position_x { get; set; } = new();
	public NoitaFloat local_position_y { get; set; } = new();
	public NoitaFloat radius_x { get; set; } = new();
	public NoitaFloat radius_y { get; set; } = new();
	public NoitaFloat capsule_x_percent { get; set; } = new();
	public NoitaFloat capsule_y_percent { get; set; } = new();
	public NoitaInt material { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		recreate.Read(s);
		is_circle.Read(s);
		is_box.Read(s);
		is_capsule.Read(s);
		is_based_on_sprite.Read(s);
		friction.Read(s);
		restitution.Read(s);
		density.Read(s);
		local_position_x.Read(s);
		local_position_y.Read(s);
		radius_x.Read(s);
		radius_y.Read(s);
		capsule_x_percent.Read(s);
		capsule_y_percent.Read(s);
		material.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		recreate.Write(s);
		is_circle.Write(s);
		is_box.Write(s);
		is_capsule.Write(s);
		is_based_on_sprite.Write(s);
		friction.Write(s);
		restitution.Write(s);
		density.Write(s);
		local_position_x.Write(s);
		local_position_y.Write(s);
		radius_x.Write(s);
		radius_y.Write(s);
		capsule_x_percent.Write(s);
		capsule_y_percent.Write(s);
		material.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/PhysicsShapeComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaInt

from NoitaStream import NoitaStream
@dataclass
class PhysicsShapeComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    recreate:NoitaBool = field(default_factory=NoitaBool)
    is_circle:NoitaBool = field(default_factory=NoitaBool)
    is_box:NoitaBool = field(default_factory=NoitaBool)
    is_capsule:NoitaBool = field(default_factory=NoitaBool)
    is_based_on_sprite:NoitaBool = field(default_factory=NoitaBool)
    friction:NoitaFloat = field(default_factory=NoitaFloat)
    restitution:NoitaFloat = field(default_factory=NoitaFloat)
    density:NoitaFloat = field(default_factory=NoitaFloat)
    local_position_x:NoitaFloat = field(default_factory=NoitaFloat)
    local_position_y:NoitaFloat = field(default_factory=NoitaFloat)
    radius_x:NoitaFloat = field(default_factory=NoitaFloat)
    radius_y:NoitaFloat = field(default_factory=NoitaFloat)
    capsule_x_percent:NoitaFloat = field(default_factory=NoitaFloat)
    capsule_y_percent:NoitaFloat = field(default_factory=NoitaFloat)
    material:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.recreate.Read(s)
        self.is_circle.Read(s)
        self.is_box.Read(s)
        self.is_capsule.Read(s)
        self.is_based_on_sprite.Read(s)
        self.friction.Read(s)
        self.restitution.Read(s)
        self.density.Read(s)
        self.local_position_x.Read(s)
        self.local_position_y.Read(s)
        self.radius_x.Read(s)
        self.radius_y.Read(s)
        self.capsule_x_percent.Read(s)
        self.capsule_y_percent.Read(s)
        self.material.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.recreate.Write(s)
        self.is_circle.Write(s)
        self.is_box.Write(s)
        self.is_capsule.Write(s)
        self.is_based_on_sprite.Write(s)
        self.friction.Write(s)
        self.restitution.Write(s)
        self.density.Write(s)
        self.local_position_x.Write(s)
        self.local_position_y.Write(s)
        self.radius_x.Write(s)
        self.radius_y.Write(s)
        self.capsule_x_percent.Write(s)
        self.capsule_y_percent.Write(s)
        self.material.Write(s)

